pkg/controller/mizar/mizar-arktos-network-controller.go

```diff
@@ -41,15 +46,32 @@ import (
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/klog"
 	"k8s.io/kubernetes/pkg/controller"
+	"k8s.io/kubernetes/pkg/features"
 )
 
 const (
-	mizarNetworkType = "mizar"
+	mizarNetworkType     = "mizar"
+	mizarInternalIPStart = 20
+
+	resource_group   = "mizar.com"
+	resource_version = "v1"
+	resource_vpc     = "vpcs"
+	resource_subnet  = "subnets"
 )
 
 var seed = rand.NewSource(time.Now().UnixNano())
 var ran = rand.New(seed)
 
+// Temporary solution for mizar VPC range cannot overlapping issue
+// VPC range will be class B ips: [vpcRangeStart.0, vpcRangeEnd.255] - cannot have overlapping across TPs
+type vpcUsedCache struct {
+	vpcRangeStart         int // Class A ip addresses start number - passed in from arg. Donot change
+	vpcRangeEnd           int // Class A ip addresses last number - passed in from arg. Donot change
+	vpcNextAvailableRange int // next VPC prefix that can be used
+	vpcUsedCache          map[int]bool
```

Comment:
in case KCM restarts, how would the ip range management work? would the allocated ranges survive the restart?

Reply:
> in case KCM restarts, how would the ip range management work? would the allocated ranges survive the restart?

populateCache() is designed for KCM restart. It was not fully tested and I found an issue [1401](https://github.com/CentaurusInfra/arktos/issues/1401). Working on it.